core/trino-main/src/test/java/io/trino/util/BenchmarkParseDate.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.RunnerException;
+
+import java.util.concurrent.TimeUnit;
+
+import static io.trino.jmh.Benchmarks.benchmark;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.openjdk.jmh.annotations.Mode.Throughput;
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+@State(Thread)
+@OutputTimeUnit(MILLISECONDS)
+@BenchmarkMode(Throughput)
+@Fork(1)
+@Warmup(iterations = 5, time = 400, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 400, timeUnit = TimeUnit.MILLISECONDS)
+public class BenchmarkParseDate
+{
+    @Benchmark
+    public void parseDate()
+    {
+        DateTimeUtils.parseDate("2023-01-11");
```

Comment:
using a compile-time constant may not be a realistic benchmark (jit can do some tricks with constants).
I would suggest generating some number of dates to parse in the @setup and iterate over them if this method.

Another thing is that the result of the `parseDate` is ignored. This can make the compiler drop the whole computation (dead code elimination). To fix this you can use `org.openjdk.jmh.infra.Blackhole` to "consume" output of every `parseDate`

Reply:
done in f19d4a7cab505376b7162119a89dc8a4169b2815